Training/Training/Training.c

```diff
@@ -67,107 +65,108 @@ int NumberReverser (int number, long long int* rev) {
 
 /// <summary>Function to drain the unassigned inputs from stdin</summary>
 void BufferDrain () {
-   char term;
-   for (; ; ) {
-      term = fgetc (stdin);
-      if (term == '\n')
-         break;
-   }
+   while (getchar () != '\n');
+}
+
+/// <summary>Function to check if the given input is a phrase or a number</summary>
+bool IsValidNum (char* phrase, int* num) {
+   char* endptr;
+   *num = strtol (phrase, &endptr, 10);
+   if (endptr == phrase || *endptr != '\0' || errno == ERANGE || phrase[0] == ' ' || phrase[0] == '\t')
+      return false;
+   else
+      return true;
 }
 
 /// <summary>Function to run test cases</summary>
 void TestCases () {
-   int strLength, numLength, numInputs[] = { 121,232,INT_MAX,INT_MIN,0 };
+   int Length, pChecker, num;

```

Comment:
The scope of these variable are limited to the for loop, so it can be declared inside the for loop itself
[https://www.geeksforgeeks.org/scope-of-a-variable/](url)